redwood-gradle-plugin/src/main/kotlin/app/cash/redwood/gradle/RedwoodSchemaPlugin.kt

```diff
@@ -52,22 +57,54 @@ public class RedwoodSchemaPlugin : Plugin<Project> {
     val compilation = kotlin.target.compilations.getByName(MAIN_COMPILATION_NAME)
     val classpath = project.configurations.getByName(compilation.compileDependencyConfigurationName)
 
-    val generate = project.tasks.register("redwoodGenerate", RedwoodSchemaTask::class.java) {
-      it.group = LifecycleBasePlugin.BUILD_GROUP
+    val generateJson = project.tasks.register("redwoodJsonGenerate", RedwoodSchemaJsonTask::class.java) {
+      it.group = BUILD_GROUP
       it.description = "Generate parsed schema JSON"
 
       it.toolClasspath.from(toolingConfiguration)
       it.outputDir.set(project.layout.buildDirectory.dir("generated/redwood"))
       it.schemaType.set(extension.type)
       it.classpath.from(classpath, compilation.output.classesDirs)
     }
+    project.tasks.named(JAR_TASK_NAME, Jar::class.java).configure {
+      it.from(generateJson)
+    }
+
+    project.afterEvaluate {
+      if (extension.apiTracking.get()) {
+        val apiFile = project.layout.projectDirectory.file("redwood-api.xml")
+
+        project.tasks.register(RedwoodApiGenerateTaskName, RedwoodSchemaApiGenerateTask::class.java) {
+          it.group = BUILD_GROUP
+          it.description = "Write an updated API tracking file for the current schema"
 
-    project.tasks.named("jar", Jar::class.java).configure {
-      it.from(generate)
+          it.toolClasspath.from(toolingConfiguration)
+          it.apiFile.set(apiFile)
+          it.schemaType.set(extension.type)
+          it.classpath.from(classpath, compilation.output.classesDirs)
+        }
+
+        val apiCheck =
+          project.tasks.register("redwoodApiCheck", RedwoodSchemaApiCheckTask::class.java) {
+            it.group = VERIFICATION_GROUP
+            it.description = "Validate the API tracking file against the latest schema"
+
+            it.toolClasspath.from(toolingConfiguration)
+            it.apiFile.set(apiFile)
+            it.schemaType.set(extension.type)
+            it.classpath.from(classpath, compilation.output.classesDirs)
+            it.dummyOutputFile.set(project.layout.buildDirectory.file("tmp/redwoodApiCheckDummy.txt"))
```

Comment:
Comment for this file? Surprises me!

Reply:
There's KDoc on the `dummyOutputFile` property inside the task. I'll move it or duplicate part of it here.